src/toxcore/tcp/codec.rs

```diff
@@ -101,6 +106,9 @@ impl Decoder for Codec {
     type Error = DecodeError;
 
     fn decode(&mut self, buf: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
+        // Add 1 to incoming counter
+        self.stats.increase_incoming();
```

Comment:
It should be incremented inside `IResult::Done` branch only.

Reply:
and same for dht codec